### PR TITLE
use sqlagg to calculate total number of rows in UCR

### PR DESCRIPTION
**Pull request by nickpell**
@NoahCarnahan @czue

### CODE REVIEW

File: corehq/apps/userreports/reports/data_source.py [hunk 1]
```diff
@@ -157,8 +158,16 @@ def has_total_row(self):
...
-        return len(self.get_data())
+        qc = self.query_context()
+        for c in self.columns:
+            qc.append_column(c.view)
+
+        session = connection_manager.get_scoped_session(self.engine_id)
+        try:
+            return qc.count(session.connection(), self.filter_values)
+        except:
+            session.rollback()
```
Comment (czue) [inline corehq/apps/userreports/reports/data_source.py]: we're just doing a select right? do we need to roll anything back?

**Comment by czue** [comment]
(not mergeable)

File: corehq/apps/userreports/reports/data_source.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/userreports/reports/data_source.py]: > we're just doing a select right?

yes

> do we need to roll anything back?

probably not, I was just following the example at https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/sqlreport.py#L273

File: corehq/apps/userreports/reports/data_source.py [hunk 2]
```diff
@@ -163,8 +164,16 @@ def has_total_row(self):
         return any(column_config.calculate_total for column_config in self.column_configs)
 
     def get_total_records(self):
-        # TODO - actually use sqlagg to get a count of rows
-        return len(self.get_data())
+        qc = self.query_context()
+        for c in self.columns:
+            qc.append_column(c.view)
```
Comment (czue) [inline corehq/apps/userreports/reports/data_source.py]: the reason we need to add all the columns is so the filters and group-by get applied, yeah? might be worth a comment (and noting that in the future we could remove the columns that are being aggregated but not part of filters or group bys)

File: corehq/apps/userreports/reports/data_source.py (hunk 1 above)
Comment (czue) [inline corehq/apps/userreports/reports/data_source.py]: interesting. it looks like the call shouldn't do anything if there's no open transaction so probably fine to leave (although could be better to remove and only add if it causes a problem of some kind)

**Comment by czue** [comment]
no blockers on my side, but just wanted to confirm i understand how this works before merging

File: corehq/apps/userreports/reports/data_source.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/userreports/reports/data_source.py]: removed try-except

File: corehq/apps/userreports/reports/data_source.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/userreports/reports/data_source.py]: that's correct.  added comment

**Comment by nickpell** [comment]
:white_check_mark:

**Comment by czue** [comment]
:+1: thanks!

**Pull request merged**